File: features/advisor/src/main/java/com/nivasafinance/features/advisor/exception/AdvisorNotFoundException.java
package com.nivasafinance.features.advisor.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import com.nivasafinance.common.exception.ResourceNotFoundException;
import org.springframework.context.MessageSource;

import java.util.UUID;

public class AdvisorNotFoundException extends ResourceNotFoundException {
    
    private static final long serialVersionUID = 1L;
    
    public AdvisorNotFoundException(UUID identifier, MessageSource messageSource) {
        super(ExceptionUtils.createLocalizedMessage(
                "error.advisor.id.not.found",
                new Object[]{identifier.toString()},
                messageSource
        ));
    }
}

File: features/advisor/src/main/java/com/nivasafinance/features/advisor/exception/AdvisorNotFoundException.kt
package com.nivasafinance.features.advisor.exception

import com.nivasafinance.common.exception.ExceptionUtils
import com.nivasafinance.common.exception.ResourceNotFoundException
import org.springframework.context.MessageSource
import java.util.UUID

class AdvisorNotFoundException(
    advisorId: UUID,
    messageSource: MessageSource
) : ResourceNotFoundException(
    ExceptionUtils.createLocalizedMessage("error.advisor.id.not.found", arrayOf(advisorId.toString()), messageSource)
)
